pkg/gardenlet/controller/networkpolicy/reconciler.go

```diff
@@ -348,11 +348,13 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToPrivateNetworks(ctx context.Co
 			return err
 		}
 
-		if v := shoot.Spec.Networking.Nodes; v != nil {
-			blockedNetworkPeers = append(blockedNetworkPeers, *v)
-		}
-		if v := shoot.Spec.Networking.Pods; v != nil {
-			blockedNetworkPeers = append(blockedNetworkPeers, *v)
+		if !shoot.IsWorkerless() {
+			if v := shoot.Spec.Networking.Nodes; v != nil {
+				blockedNetworkPeers = append(blockedNetworkPeers, *v)
+			}
+			if v := shoot.Spec.Networking.Pods; v != nil {
+				blockedNetworkPeers = append(blockedNetworkPeers, *v)
+			}
 		}
 		if v := shoot.Spec.Networking.Services; v != nil {
```

Comment:
Nil check for networking is missing.

Reply:
Networking will never be nil for Shoot with workers, I can still add the check.